src/devices/QwiicButton/README.md

```diff
@@ -0,0 +1,124 @@
+﻿# SparkFun Qwiic Button
+
+## Summary
+
+The [SparkFun](https://www.sparkfun.com) Qwiic Button is an [I²C](https://en.wikipedia.org/wiki/I%C2%B2C) based button with a built-in LED.
+
+In addition to handling blinking and debouncing, the Qwiic Button has configurable interrupts that can be configured to activate upon a button press or click. The Qwiic Button also contains a FIFO queue where it keeps an internal record of when the button was pressed.
+
+![Qwiic Button](./samples/15932-SparkFun_Qwiic_Button_-_Red-01.png)
+
+At a glance:
+
+-   12mm LED button rated for 50mA
+-   Configurable LED blinking
+-   Configurable I²C address
+-   Configurable interrupts
+-   FIFO queue
+
+## Device Family
+
+Compatible with devices capable of acting as I²C master, such as the [Raspberry Pi](https://www.raspberrypi.org).
+
+## Fritzing Diagram
+
+The following [Fritzing](https://fritzing.org) diagram shows how to connect a Raspberry Pi 4 to the Qwiic Button:
+
+![Fritzing Diagram](./samples/QwiicButton-RPi4-Fritzing.png)
+
+## Binding Notes
+
+This binding is a port of SparkFun's [Qwiic Button Arduino Library](https://github.com/sparkfun/SparkFun_Qwiic_Button_Arduino_Library) (MIT licensed).
+
+The supported hardware version is 1.0.0.
+
+The functionality of the button is demonstrated in the samples. Below are code samples of some common usage scenarios.
+
+### Detect Button Is Pressed Using Polling
+
+The method `QwiicButton.IsPressed()` indicates whether the button has been pressed:
+
+```c#
+using (var button = new QwiicButton(i2cBusId: 1, i2cAddress: 111))
+{
+    while (!Console.KeyAvailable)
+    {
+        // Check if button is pressed, and tell us if it is!
+        if (button.IsPressed())
+        {
+            Console.WriteLine("The button is pressed!");
+            while (button.IsPressed())
+            {
+                Thread.Sleep(10); // Wait for user to stop pressing
+            }
+
+            Console.WriteLine("The button is not pressed anymore!");
+        }
+
+        Thread.Sleep(20); // Don't hammer too hard on the I2C bus
+    }
+}
+```
+
+### Detect Button Is Pressed Using an Interrupt
+
+Interrupts are handled by a `GpioController` instance that detects changes on a GPIO pin.
+Below is an example that uses the Raspberry Pi's GPIO pin 23:
+
+```c#
+private QwiicButton _button;
+
+public void Run()
+{
+    using (_button = new QwiicButton(i2cBusId: 1, i2cAddress: 111))
+    {
+        Initialize();
+        Console.ReadKey(); // Block the thread and wait for OnValueChanged to be invoked
+    }
+}
+
+private void Initialize()
+{
+    var gpioController = new GpioController(PinNumberingScheme.Board, new RaspberryPi3Driver());
+    gpioController.OpenPin(pinNumber: 23, PinMode.Input);
+    gpioController.RegisterCallbackForPinValueChangedEvent(
```

Comment:
is this working fine for you when you're testing? did you see any missed events?

Reply:
On a rare occasion, I might have seen an event not firing, but most of the time it works just fine.